libs/ml/include/ml/ops/ops.hpp

```diff
@@ -149,6 +151,60 @@ class Ops
     // Empty deafult implementation for non-trainable Ops.
   }
 
+  /**
+   * @brief ChargeForward
+   * @return estimated charge amount, necessary for performing a forward pass on data of given
+   * shapes.
+   */
+  virtual MLChargeAmount ChargeForward()
+  {
+    // TODO(VH): make a pure virtual call;
```

Comment:
so, why don't you just make this pure virtual like your todo suggests?

Reply:
this require immedtiate implementation across all Ops, which is huge PR; so I'd do it when all Ops would have their overriden implementation